src/Microsoft.Health.Blob/Features/Storage/BlobContainerInitializer.cs

```diff
@@ -25,14 +30,22 @@ public BlobContainerInitializer(string containerName, ILogger<BlobContainerIniti
         }
 
         /// <inheritdoc />
-        public async Task<BlobContainerClient> InitializeContainerAsync(BlobServiceClient client)
+        public async Task<BlobContainerClient> InitializeContainerAsync(BlobServiceClient client, CancellationToken cancellationToken = default)
         {
             EnsureArg.IsNotNull(client, nameof(client));
 
             BlobContainerClient container = client.GetBlobContainerClient(_containerName);
 
-            _logger.LogDebug("Creating blob container if not exits: {containerName}", _containerName);
-            await container.CreateIfNotExistsAsync();
+            _logger.LogDebug("Creating blob container if not exists: {containerName}", _containerName);
+
+            try
+            {
+                await container.CreateIfNotExistsAsync(cancellationToken: cancellationToken);
+            }
+            catch (OperationCanceledException ex)
+            {
+                _logger.LogCritical(ex, "Blob container initialization cancelled: {containerName}", _containerName);
```

Comment:
Same comment as the other file. Should we log this or simply log at the top of the call stack in the hosted service.

Reply:
fixed